### PR TITLE
Vuln scan weekly

### PR DESCRIPTION
**Pull request by sehaartuc**
Hello, I noticed an error while fetching & sorting for the latest two tags and fixed it. For your information.

### CODE REVIEW

File: .github/workflows/weekly_vuln_scan.yml [hunk 1]
```diff
@@ -1,41 +1,60 @@
...
   schedule:
-    - cron: "0 14 * * 1" # Every Monday at 14:00 UTC
+    - cron: "0 14 * * 1" # Works on Every Monday at 14:00 UTC
   workflow_dispatch:
 
 jobs:
   fetch-and-scan:
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout repository
```
Comment (brunopacheco1) [inline .github/workflows/weekly_vuln_scan.yml:15]: question: why do you need to do checkout of the repo, if we are scanning only docker images?

File: .github/workflows/weekly_vuln_scan.yml [hunk 2]
```diff
@@ -1,41 +1,60 @@
...
+          echo "Response from curl: $RESPONSE"
+
+          # Extract tags and creation dates, then sort by creation date
+          TAGS_WITH_DATES=$(echo "$RESPONSE" | jq -r '.[] | .metadata.container.tags[] as $tag | "\($tag) \(.created_at)"' | sort -k2 -r)
+
+          # Filter out the 'main' tag and get the latest two tags
+          LATEST_TAGS=$(echo "$TAGS_WITH_DATES" | awk '$1 != "main" {print $1}' | head -n 2 | tr '\n' ' ')
+
+          # Print the results
+          echo "Tags fetched: $LATEST_TAGS"
```
Comment (brunopacheco1) [inline .github/workflows/weekly_vuln_scan.yml:44]: question: why did you need to fix this algorithm?

We need to check only the latest from the current version and the latest from the previous version. 

Fetching the last two sorted by created date will not fulfil this requirement, as it may fetch the last 2 of the same version.

File: .github/workflows/weekly_vuln_scan.yml [hunk 3]
```diff
@@ -1,41 +1,60 @@
...
   fetch-and-scan:
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
       - name: Setup Docker
         run: |
           sudo service docker start
+          docker ps
```
Comment (brunopacheco1) [inline .github/workflows/weekly_vuln_scan.yml:21]: question: why do you need show the processes?

File: .github/workflows/weekly_vuln_scan.yml (hunk 1 above)
Comment (sehaartuc) [reply .github/workflows/weekly_vuln_scan.yml:15]: It's not necessary actually, I've added it just because if any case needs repo based analysis. However it's not necessary, let me remove this part.

File: .github/workflows/weekly_vuln_scan.yml (hunk 3 above)
Comment (sehaartuc) [reply .github/workflows/weekly_vuln_scan.yml:21]: It's only for understanding of docker daemon is working or not but I can remove it also.

**Review by brunopacheco1 — DISMISSED** [review]

**Review by brunopacheco1 — APPROVED** [review]
